coreutils/whoami.rs

```diff
@@ -35,14 +24,13 @@ use libc::uid_t;
 //usage:#define whoami_full_usage "\n\n"
 //usage:       "Print the user name associated with the current effective user id"
 #[no_mangle]
-pub unsafe extern "C" fn whoami_main(
-  mut _argc: libc::c_int,
-  mut argv: *mut *mut libc::c_char,
-) -> libc::c_int {
+pub unsafe extern "C" fn whoami_main(mut _argc: c_int, mut argv: *mut *mut c_char) -> c_int {
   if !(*argv.offset(1)).is_null() {
     bb_show_usage();
   }
   /* Will complain and die if username not found */
-  puts(xuid2uname(geteuid()));
-  return fflush_all();
+  let euid = geteuid();
+  let user_name = xuid2uname(euid);
+  println!("{}", *user_name);
+  fflush_all()
```

Comment:
Do you think we can get away without this? I imagine everything should still flush before exiting... If not we could always `std::io::stdout().flush()`.